Major refactor of connection code

theJenix opened the pull request:
Connection code from HostActivity and FanActivity pulled into ConnectionService and MessagingService.

All connection management code now lives in ConnectionService.  This includes
bluetooth discovery, host/fan connection management, and sending and receiving
messages.

All messaging management code now lives in MessagingServices.  It receives new
IMessage objects from the ConnectionService, decomposes the messages into their
parts, and sends a broadcast intent/action to the rest of the system to
represent each message.  This decouples the message and network layer from the
rest of the app.

HostActivity and FanActivity have been modified to use these services.

Currently broken: remote fan discovery.  Not too far from being complete...

theJenix committed:
Major refactor of connection code.  Connection code from HostActivity and
FanActivity pulled into ConnectionService and MessagingService.

All connection management code now lives in ConnectionService.  This includes
bluetooth discovery, host/fan connection management, and sending and receiving
messages.

All messaging management code now lives in MessagingServices.  It receives new
IMessage objects from the ConnectionService, decomposes the messages into their
parts, and sends a broadcast intent/action to the rest of the system to
represent each message.  This decouples the message and network layer from the
rest of the app.

HostActivity and FanActivity have been modified to use these services.

Currently broken: remote fan discovery.  Not too far from being complete...

ejohnson44 commented on fanClub/src/com/lastcrusade/fanclub/FanActivity.java:
```diff
@@ -98,96 +72,87 @@ public void onClick(View v) {
-        this.messageThread = new MessageThread(socket, this.messageDispatch);
-        this.messageThread.start();
-    }
+//    private void handleFoundFans(List<FoundFan> list) {
```
Why is all of this commented out?

theJenix replied:
that's the part thats not quite working yet (remote display of found fans).  If it's agreeable to everyone, my plan would be to pull this in, get it all integrated, and then revisit the remote finding of fans afterward.

bryansills commented on fanClub/src/com/lastcrusade/fanclub/FanActivity.java:
```diff
@@ -98,96 +72,87 @@ public void onClick(View v) {
+                    Toaster.iToast(FanActivity.this, string);
+                }
+            })
+//            .addAction(ConnectionService.ACTION_FIND_FINISHED, new IBroadcastActionHandler() {
```
why is this commented?

theJenix replied:
See above comment...this is part of the comment that isnt working yet.

theJenix committed:
Added Play/Pause/Skip messages and unbind call in onDestroy

theJenix committed:
Added comment to the top of ServiceLocator, talking about pitfalls when using
services.

Changed Runnable to a more specific interface, for on bind listeners.

theJenix committed:
Fixed StringMessage sending to support both fans and hosts
